dashboard/app/jobs/mail_delivery_job.rb

```diff
@@ -8,4 +10,16 @@
 # @see https://apidock.com/rails/v6.1.7.7/ActionMailer/MessageDelivery/deliver_later
 class MailDeliveryJob < ActionMailer::MailDeliveryJob
   include ActiveJobMetrics
+
+  rescue_from StandardError, with: :report_exception
```

Comment:
NOTE: The mail exceptions will be reported without retrying the job

Reply:
If we have a problem and a bunch of email can't get sent, how should we go about re-queuing up those messages?

Reply:
The same as you re-queuing up regular failed jobs

Reply:
But I believe we currently delete failed jobs: https://github.com/collectiveidea/delayed_job/blob/b66bb64437a8606a414a390531ac73108911434e/README.md?plain=1#L444-L445

Reply:
Yes, I believe we currently delete failed jobs. We might want to re-think that.

Reply:
We could also add Web UI for managing jobs: https://github.com/ejschmitt/delayed_job_web

Reply:
https://github.com/code-dot-org/code-dot-org/pull/58093/commits/8e230818025c4c183ea5319e99556c42b5fde500